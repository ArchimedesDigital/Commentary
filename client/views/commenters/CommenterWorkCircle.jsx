
CommenterWorkCircle = React.createClass({

	propTypes: {
		toggleVisibleWork: React.PropTypes.func.isRequired,
		workTitle: React.PropTypes.string.isRequired,
		workSlug: React.PropTypes.string.isRequired,
		workLevel: React.PropTypes.number.isRequired,
	},

	render() {
		return (
			<div className={`commenter-work-circle circle-level-${this.props.workLevel}`}>
				<div
					className="circle-inner"
					onClick={this.props.toggleVisibleWork.bind(null, this.props.workSlug)}
				>
					<span className="work-title">{this.props.workTitle}</span>
					<div className="grow-border" />
				</div>
			</div>
		);
	},
});
