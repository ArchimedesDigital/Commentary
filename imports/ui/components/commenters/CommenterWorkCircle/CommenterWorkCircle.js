import React from 'react';

const CommenterWorkCircle = (props) => (

	<div className="commenter-work-circle">
		<div
			className={`circle-inner circle-level-${props.workLevel}`}
			onClick={props.toggleVisibleWork.bind(null, props.workSlug)}
		>
			<span className="work-title">{props.workTitle}</span>
			<span className="work-count">{props.nComments}</span>
			<div className="grow-border" />
		</div>
	</div>
);


CommenterWorkCircle.propTypes = {
	toggleVisibleWork: React.PropTypes.func.isRequired,
	workTitle: React.PropTypes.string.isRequired,
	workSlug: React.PropTypes.string.isRequired,
	workLevel: React.PropTypes.number.isRequired,
	nComments: React.PropTypes.number.isRequired,
};

export default CommenterWorkCircle;
