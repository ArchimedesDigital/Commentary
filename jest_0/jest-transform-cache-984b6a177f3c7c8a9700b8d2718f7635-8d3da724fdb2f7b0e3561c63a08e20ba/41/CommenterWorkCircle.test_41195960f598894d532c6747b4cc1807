/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommenterWorkCircle = require('./CommenterWorkCircle'); /* istanbul ignore next */var _CommenterWorkCircle2 = _interopRequireDefault(_CommenterWorkCircle);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommenterWorkCircle', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommenterWorkCircle2.default, /* istanbul ignore next */{
			workTitle: 'Iliad',
			workSlug: 'iliad',
			workLevel: 0,
			nComments: 0,
			toggleVisibleWork: function /* istanbul ignore next */toggleVisibleWork() {} })).


		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: