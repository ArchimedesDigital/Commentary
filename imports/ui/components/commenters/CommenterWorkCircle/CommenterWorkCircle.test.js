import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommenterWorkCircle from './CommenterWorkCircle';

describe('CommenterWorkCircle', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<CommenterWorkCircle
					workTitle="Iliad"
					workSlug="iliad"
					workLevel={0}
					nComments={0}
					toggleVisibleWork={() => {}}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
