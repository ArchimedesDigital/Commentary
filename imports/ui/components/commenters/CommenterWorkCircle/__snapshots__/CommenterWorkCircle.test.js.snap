// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommenterWorkCircle renders correctly 1`] = `
<div
  className="commenter-work-circle"
>
  <div
    className="circle-inner circle-level-0"
    onClick={[Function]}
  >
    <span
      className="work-title"
    >
      Iliad
    </span>
    <span
      className="work-count"
    >
      0
    </span>
    <div
      className="grow-border"
    />
  </div>
</div>
`;
